lib/core/src/persist/mod.rs

```diff
@@ -395,6 +396,13 @@ fn filter_to_where_clause(
     where_clause.join(" and ")
 }
 
+fn hash_str(s: &str) -> String {
+    let mut hasher = Sha256::new();
```

Comment:
I would not add a dependency on openssl, as it's generally bloated and will make it harder if we want to switch from it in the future.

You can use the sha256 impl from the bitcoin crate instead, like we do for building the MRH:

```rust
let hash_bytes = sha256::Hash::hash(swap.id.as_bytes());
let hash_str = hash_bytes.to_byte_array().to_hex();
```